### PR TITLE
Only select and read markdown files for query parsing

### PR DESCRIPTION
**Pull request by nbennett320**

### CODE REVIEW

File: packages/source-filesystem/src/index.ts [hunk 1]
```diff
@@ -5,27 +5,37 @@ import { read } from 'to-vfile';
...
+ * @param dir The directory to read from
+ * @param extensions File extensions to parse
+ * @returns A list of content nodes' filenames
+ */
+export async function getValidNodesFilenames(
+  dir: string,
+  extensions?: string[]
+): Promise<string[]> {
+  // If no extensions are provided, use the default ones
+  let validExtensions = extensions ?? ['.md', '.mdx', '.markdown'];
```
Comment (tonyketcham) [inline packages/source-filesystem/src/index.ts:21]: this should be const

**Comment by tonyketcham** [comment]
Thank you for your service

**Comment by nbennett320** [comment]
Do you want to squash branch merges

**Comment by nbennett320** [comment]
This verifies markdown files properly but will still pick up hidden files/files that start `.like_this`. Is it intended..? Does it matter..?

**Comment by tonyketcham** [comment]
That's an interesting point. I don't know how often someone would come across a hidden file of the same type as the ones being parsed, though I could be underthinking the applications people would use this thing for. 

A niche but potential case that could occur is when reading a directory of existing relational flat files where someone has chosen to hide files intentionally, like a `.LICENSE.md`. A few thoughts around adding support for excluding hidden files: 
1. Audience & use cases - would that feature satisfy the 80/20 rule?
2. Would it add more overhead than it's worth? 
    a. Moreover, could this be disabled by default, and those who wanna use it can opt-in?
2. Would it cause any false-positives, like if a flat-file CMS slugifies the file name with a period at the beginning?

**Comment by tonyketcham** [comment]
Following up for the sake of continuity -- @nbennett320 brought up some good points outside of this chain and I've been convinced that excluding hidden files would be a good idea with an advanced config option to switch that check on or off

**Pull request merged**